### PR TITLE
[Shape] Add isEqual support to our corner treatments

### PR DESCRIPTION
**Pull request by yarneo**
This will make it easier to compare between corner treatments for testing purposes among other things.

### CODE REVIEW

**Review by codeman7 — APPROVED** [review]

**Review by jverkoey — DISMISSED** [review]
Add tests please.

File: components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#import <XCTest/XCTest.h>
+
+#import "MDCCornerTreatment+CornerTypeInitalizer.h"
+#import "MDCCurvedCornerTreatment.h"
+#import "MDCCurvedRectShapeGenerator.h"
+#import "MDCCutCornerTreatment.h"
+#import "MDCPillShapeGenerator.h"
+#import "MDCRoundedCornerTreatment.h"
+#import "MDCSlantedRectShapeGenerator.h"
+#import "MDCTriangleEdgeTreatment.h"
```
Comment (jverkoey) [inline components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m:24]: `#import "MaterialShapeLibrary.h"`

File: components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m [hunk 2]
```diff
@@ -0,0 +1,125 @@
+// Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (jverkoey) [inline components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m:1]: Copyright year is 2018.

File: components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      [[MDCCornerTreatment alloc] initWithCornerType:MDCCornerTypeRounded andSize:value];
+
+  // When
+  XCTAssertNotEqualObjects(roundedCorner, roundedCorner2);
+  roundedCorner2.radius = 3.2f;
+
+  // Then
+  XCTAssertEqualObjects(roundedCorner, cornerTreatment);
+  XCTAssertEqualObjects(roundedCorner, roundedCorner2);
+  XCTAssertEqualObjects(roundedCorner, roundedCorner);
```
Comment (jverkoey) [inline components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m:69]: Consider splitting the equality and inequality tests apart to increase readability.

**Review by jverkoey — COMMENTED** [review]
Ahhh I forgot to submit my review feedback :(

**Pull request merged**